### DISCUSSION TITLE
Why is there no Jamulus Radio? and various confusions

### DISCUSSION THREAD

**Discussion by RockTribe** [jamulussoftware/jamulus#2816] [Ideas]
Greetings, I have been busy with online improvisation music for two years.

My connection is not enough for Jamulus. I'm getting around minimum 60ms of latency and the sound is constantly stuttering.

I asked my friends who connected from Turkey, they also get the same latency, but they continue to use it.

Is there anything we can do to the audio problems? Or does everyone who uses it experience these problems?

Also why can't we listen to the music from jamulus.live? Or why isn't there a similar radio page? I ran into a few servers. They were broadcasting live with sites that turned it into audio stream. I've also heard sound problems with them. This made me suspect that this is the case with everyone.

I understand that many people who use ninjam are now using jamulus. Honestly, I'm pretty happy with the ninjam. Whether it's real time or not doesn't make a big difference when playing on loops. If I can use jamulus properly, I'll play it on loops as well, but first of all, the sound should be smooth (as in ninjam) and we should be able to stream our work easily.

Thank you, have a good day.

**Comment by chrisrimple** [comment]
In what country are you located?  If you're anywhere in Europe, you'll have the best connections to servers in Europe, not North America.  60ms Overall Delay isn't too bad, and you should be hearing sound without stuttering.  In the Jamulus window, do you see your Jitter light come on red a lot?  If so, that's a problem on your end, not the Jamulus server.  Jitter can happen when there are too many users on your internet connection - be sure that Wifi is disabled at your router, so your computer running Jamulus is getting your internet connection all to itself.

Jamulus is not intended to be a streaming radio service - it's just a tool for musicians to play together via the internet.  If someone wants to login to a server and play a continuous selection of music into a Jamulus client, that would be more like radio.  But the only people that could hear it would be other Jamulus users which join that server, so it's not a method for putting out a radio stream.  Better to stream live via YouTube, Twitch, etc.  That's why you can't listen from Jamulus.live - the Jamulus server is not outputting a "public stream", it's only putting out a "private stream" to connected Jamulus clients.

**Reply by RockTribe** [reply]
I'm from Turkey, I asked my friends who use jamulus from Turkey and they said they had similar problems.

The jitter light is not always on, but when it is on, there is of course distortion in the sound. I am aware that the quality of my connection is poor. however, I have seen and heard the sound problems I mentioned both in friends who posted it on their site and in youtube videos.

So you're saying I should be able to use it with 60ms with no issues with sound?

I don't understand the politics about radio at all. It is not designed that way, but they can create an opportunity this way if they want.

I think it would be nice if we could click and listen, wondering what people are playing.

This would also be nice for the performers because of the possibility of being in the audience.

--
by the way nothing is listed on jamulus application for example "any 1" server. When I searched about it, they says there may be a problem with the connection, security (or port) settings, but is it related to the problem I mentioned above? I do not know.

**Comment by chrisrimple** [comment]
Yes, you should be able to use it at 60ms Overall Delay without sound issues.  On the [WorldJam](https://www.youtube.com/c/WorldJam), musicians have played with up to 120ms Overall Delay and, while staying synchronized was difficult, there were no sound issues.  If you're hearing distortion in the sound, it's probably not the server - more likely, it's the individual musicians (sending distorted sound to the server) or the local playback (your CPU is overloaded, you have an audio driver problem, etc.).

On the subject of radio, which is really the subject of a "broadcast mode" for the Jamulus Server, that would need to be built by the developers (who are all volunteers), OR it could be accomplished by having someone run the Jamulus Client and redirect the audio output to a streaming service (YouTube, etc.).  That's how the WorldJam does it for their shows.  But to enable that on Jamulus.live would require (1) a broadcast mode for the Jamulus Server and (2) website code that could receive and play the stream (to install on Jamulus.live and elsewhere).  

Finally, I think you're saying that, when you are in the Jamulus Client and you choose Any Genre, you don't see any servers listed?  If that's true, it sounds like (1) you have a connection problem or (2) a configuration problem.  You might try deleting the jamulus.ini file and reinstalling, to ensure that you have the current list of Directory Servers.

**Reply by RockTribe** [reply]
Thanks for the reply. I tried again last night. I don't have a problem with 60ms when playing alone, but even if one person is connected, my sound is slightly distorted. The jitter light wasn't on either.

This problem gets worse when you lower the buffer setting. It decreases when you set it to the highest.

I am using it on mac. I re-set the privacy settings, the connection is not blocked. I can't see any1 and any2 lists, the others are fine.

I will try to fix it over the ini file you mentioned.

I hope there will be an improvement on the radio. Especially people with an upload speed of 0.6mbps like me cannot broadcast to two places at the same time (It is not useful for people to broadcast one by one anyway). Unfortunately, the speeds are like this in Turkey, I have to use this connection for at least 1 more year.

I think this online music topic needs a community. Of course, those who want to play on their own will not participate, but there are many people who follow such content and produce content. We can make good music together, but for this we need to be able to listen first and then reach each other.

I couldn't find a discord channel for jamulus either. other programs have a discord channel, but there seems to be no active use...

**Reply by pljones** [reply]
> I can't see any1 and any2 lists, the others are fine.

That implies somewhere between you and the directory there is a router with fragmentation of large UDP packets and they're not being handled correctly.  Which means you could be experiencing other network issues.  However, it's really hard to diagnose any problems without data.

**Reply by ann0see** [reply]
> I think this online music topic needs a community. Of course, those who want to play on their own will not participate, but there are many people who follow such content and produce content. We can make good music together, but for this we need to be able to listen first and then reach each other.

> I couldn't find a discord channel for jamulus either. other programs have a discord channel, but there seems to be no active use...

The community is hidden inside Jamulus ;-), and yes, that's a hurdle. We don't have any official Discord server, but there's a Facebook group: https://www.facebook.com/groups/619274602254947/

**Reply by ann0see** [reply]
Ah yes, and there's also one on reddit, I don't know how active they are and who maintains it: https://www.reddit.com/r/Jamulus/

**Reply by RockTribe** [reply]
> The community is hidden inside Jamulus ;-), and yes, that's a hurdle. We don't have any official Discord server, but there's a Facebook group: https://www.facebook.com/groups/619274602254947/

no, it's not like that :) Facebook doesn't work very well in this regard.

For example, in jamulus, there are people who connect only to listen, their nicknames are "listening".

While I'm out, I can listen to who's playing what on the ninbot.com site. Something like this would be very useful.

 Beyond that, it could be a site that combines the chats of all online music playing sites. It could be like the lobby of the old style chat pages.

Who would connect to which system, this could be seen and listened to, and an instant message could be sent if necessary.

In other words, being able to listen to music and being accessible are the most necessary things for the community. Unfortunately, these are not available in Jamulus, you need to be connected to listen, a computer is required and we cannot reach outside and send messages.

Ninjam, jamulus, sonobus, jammr, jacktrip etc, i guess they all use some kind of chat, I don't know if it is possible to make an "online music lobby" site that will cover them all, but if there were, it would be very useful for me.

**Reply by pljones** [reply]
> While I'm out, I can listen to who's playing what on the ninbot.com site. Something like this would be very useful.

That's because there's a client called "ninbot" connected to the server, and that's got a feed out to a streaming program.  There are servers on Jamulus that do provide the same service.  There's nothing official about ninbot.com - nor any other NINJAM server.  They're just registered with a single central "directory" equivalent and you'll find not all of them have streaming, as not all are run by people who want to get into providing such a service.

> Beyond that, it could be a site that combines the chats of all online music playing sites. It could be like the lobby of the old style chat pages.

As with streaming services, that's always going to an "opt in" thing.  If a server operator wants to mention such a service in their welcome message, then they could, with a link.  It's never going to be forced on a server operator.  So you'll always be in a position where the server you want to be "in" is operated by someone who thinks they should be "out"...

> Unfortunately, these are not available in Jamulus

As they're not about sending audio to the server and getting the mix back.

> Ninjam, jamulus, sonobus, jammr, jacktrip etc, i guess they all use some kind of chat, I don't know if it is possible to make an "online music lobby" site that will cover them all, but if there were, it would be very useful for me.

Jamulus has an RPC client interface.  Someone could use that to connect to any server as a user and monitor chat and stream it out somewhere else.  Whether _any other service_ does the same I don't know.  But that's not a fault in Jamulus.

**Reply by ann0see** [reply]
@RockTribe I think you want to lower the hurdle for beginners and a central place for getting to know people? Is it that? 

The music lobby idea was already somewhat implemented (lounge.jamulus.io Jamulus.live and so on). It’s certainly possible to set up a service like that but someone needs to make it/develop it further.

Another option would be a discord Server or something comparable (with bots streaming live to voice channels). However, everything like that needs moderators, community engagement and should be at one central place to bring the community together. This would then not be "officially supported" but everyone can start that.

**Reply by ann0see** [reply]
> While I'm out, I can listen to who's playing what on the ninbot.com site. Something like this would be very useful.

Unless you use an exotic or old mobile device, iOS and Android have working clients. iOS works better, but the App Store version is not up to date and we don’t know the maintainers, but it "works fine for me". Android depends on the device. 
***
Please also keep in mind that some server owners and users (including myself) want to know who’s listening/reading chat messages.

**Reply by ann0see** [reply]
@RockTribe I just found https://github.com/jamulussoftware/jamulus/discussions/2615 again. Please comment there.

**Comment by RockTribe** [comment]
I found an alternative when I was searching for such a natural and logical feature. I haven't tried if it works, but if anyone wants to try:
https://www.jacktrip.com/jacktrip-radio

**Comment by mcfnord** [comment]
> Also why can't we listen to the music from jamulus.live? 

I do like this idea. My own idea is much smaller: Short, recent audio samples. I think that might help people unfamiliar with Jamulus get a little taste before installing anything, and also help users decide where to go next. But streaming 24-7 from every server isn't realistic for reasons people have mentioned. Maybe someday!
